tasks/main.yml

```
@@ -117,32 +123,46 @@
     - ansible_distribution in ['CentOS', 'RedHat']
     - ansible_distribution_version is version('9', '>=')
 
-- name: Join to a specific Domain Controller
-  shell: |
-    set -euo pipefail
-    echo {{ ad_integration_password | quote }} | realm join -U \
-      {{ ad_integration_user | quote }} --membership-software \
-      {{ ad_integration_membership_software | quote }} \
-      {{ ad_integration_join_to_dc | quote }}
+- name: Build Command - Join to a specific Domain Controller
+  set_fact:
+    join_command: |
```

```suggestion
    __ad_integration_join_command: |
```
Variables set with `set_fact` are global variables visible to everything in the playbook.  We don't have a mechanism in Ansible to "protect" variables like this, so we have to use a naming convention.

---

Done.